Modules/_winapi.c

```diff
@@ -1699,6 +1699,80 @@ _winapi_GetFileType_impl(PyObject *module, HANDLE handle)
     return result;
 }
 
+/*[clinic input]
+_winapi.CreateJobObject -> HANDLE
+
+    limit_info: object
+    job_name: Py_UNICODE(accept={str, NoneType}) = NULL
+    /
+
+Creates and returns the handle for a new Job Object.
+
+[clinic start generated code]*/
+
+static HANDLE
+_winapi_CreateJobObject_impl(PyObject *module, PyObject *limit_info,
+                             Py_UNICODE *job_name)
+/*[clinic end generated code: output=ddb90fccf4f363b9 input=f1350a07abdab61c]*/
+{
+    HANDLE job_handle;
+    Py_BEGIN_ALLOW_THREADS
+    job_handle = CreateJobObjectW(NULL, job_name);
+    Py_END_ALLOW_THREADS
+
+    if (job_handle == NULL) {
+        PyErr_SetFromWindowsErr(GetLastError());
+        return INVALID_HANDLE_VALUE;
+    }
+
+    JOBOBJECT_EXTENDED_LIMIT_INFORMATION limit;
+    ZeroMemory(&limit, sizeof(limit));
+    // we only support a subset of all the limits possible, can be expanded
+    // in the future if needed.
+    limit.BasicLimitInformation.LimitFlags =
+        (DWORD) getulong(limit_info, "LimitFlags");
```

Comment:
Maybe this should reflect the WinAPI structure, i.e. the `BasicLimitInformation` field. Also, it would be nice to have PyWin32 interoperability, which does the latter and also uses dicts instead of simple namespaces.

Reply:
Hmm, so I based this on `CreateProcess` which also uses attributes in a passed in object for this information. What do you mean by reflect the structure? Do you mean support all the fields for `BasicLimitInformation`?

Reply:
For reference, here is how subprocess passes uses it with `CreateProcess` https://github.com/python/cpython/blob/06ca3f0c09d017b9d741553818459cca2d5da587/Lib/subprocess.py#L129-L137

I just used simple namespace because I was being lazy

Reply:
Sometimes a dict is used, e.g. `STARTUPINFO` uses a dict for the new `lpAttributeList` support in 3.7. `win32job` uses dicts, but it's fine to stick with namespaces instead. I still would rather `LimitFlags` be set in a `BasicLimitInformation` attribute instead of at the top level. _winapi is low-level and undocumented, so it's best if it mirrors the actual API, if it's not overly awkward. I didn't meant to support all fields, however. _winapi gets extended only as required.